src/components/Connect.vue

```diff
@@ -1,81 +1,39 @@
 <script setup lang="ts">
-import { getState, useState } from '@/stores'
+import { getState, useState, setState } from '@/stores'
 import { ref, onMounted, onUnmounted, watch } from 'vue'
-import { provider as Provider } from 'web3-core'
-import { EthereumProviderError } from 'eth-rpc-errors'
 import useDropdown from '@/compositions/useDropdown'
-import useWeb3 from '@/compositions/useWeb3'
-import {
-  MEANS_OF_CONNECTION,
-  UP_CONNECTED_ADDRESS,
-  WALLET_CONNECT,
-  WINDOW_ETHEREUM,
-} from '@/helpers/config'
-import useWalletConnectV2 from '@/compositions/useWalletConnectV2'
+import useWeb3Connection from '@/compositions/useWeb3Connection'
+import { WALLET_CONNECT, WEB3_ONBOARD, WINDOW_LUKSO } from '@/helpers/config'
+import EthereumProvider from '@walletconnect/ethereum-provider/dist/types/EthereumProvider'
 import { sliceAddress } from '@/utils/sliceAddress'
 
-const { setupWeb3, accounts, requestAccounts } = useWeb3()
+const { setupProvider, disconnect } = useWeb3Connection()
 
-const { resetWCV2Provider, setupWCV2Provider, openWCV2Modal } =
-  useWalletConnectV2()
-const { setDisconnected, setConnected } = useState()
 const { close, toggle } = useDropdown()
 const dropdown = ref()
-const hasExtension = ref<boolean>(!!window.ethereum)
+const hasExtension = ref(false)
+const provider = ref<EthereumProvider>()
 
 watch(
-  () => !!window.ethereum,
+  () => !!window.lukso,
```

Comment:
We still need to be able to allow metamask to connect, so it can not be hardcoded to window.lukso. The easiest would be to continue to check window.ethereum here since both mm and up extension also inject into window.ethereum.

A more robust solution would be to return a boolean `extensionInstalled` property from useWeb3Connection() and do the check there.

Reply:
That check is to see if we have the Lukso extension available. I dont know if we want to also check if MM is available. Its a UI text displayed to see if UP is installed.

<img width="1430" alt="image" src="https://github.com/lukso-network/universalprofile-test-dapp/assets/44970013/4d3c3cb6-ae58-4c04-abd9-ac3f78324120">